web/pdf_print_service.js

```diff
@@ -27,8 +29,28 @@ function renderPage(
   pageNumber,
   size,
   printResolution,
-  optionalContentConfigPromise
+  optionalContentConfigPromise,
+  printContainer
 ) {
+  if (pdfDocument.isPureXfa) {
+    const page = document.createElement("div");
+    page.setAttribute("class", "xfaPrintedPage");
+    printContainer.appendChild(page);
+
+    return pdfDocument.getPage(pageNumber).then(pdfPage => {
+      const builder = new DefaultXfaLayerFactory().createXfaLayerBuilder(
+        page,
+        pdfPage,
+        pdfDocument.annotationStorage
+      );
+      const viewport = pdfPage.getViewport({
+        scale: CSS_UNITS,
+        rotation: size.rotation,
+      });
+      return builder.render(viewport, "print").promise;
```

Comment:
I know that this is still a WIP, but for future reference:
As far as I can tell, the `intent` parameter is currently being ignored in the `render` method.